src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -843,8 +843,17 @@ protected void StoreInReference (MultiValue target, MultiValue source, MethodDef
 					// Ref returns don't have special ReferenceValue values, so assume if the target here is a MethodReturnValue then it must be a ref return value
 					HandleStoreMethodReturnValue (method, methodReturnValue, operation, source);
 					break;
+				case IValueWithStaticType typedValue:
+					if (typedValue.StaticType is not TypeReference typeRef
+						|| ResolveToTypeDefinition (typeRef) is not TypeDefinition typeOfTarget)
+						break;
+
+					if (typeOfTarget.IsTypeOf(WellKnownType.System_String) || typeOfTarget.IsTypeOf (WellKnownType.System_Type))
```

Comment:
Please make the method on FlowAnnotations internal and call it from here - we don't want to duplicate the same logic.